src/Fantomas.Core/CodePrinter.fs

```diff
@@ -3212,6 +3212,8 @@ let genTypeDefn (td: TypeDefn) =
         |> genNode typeName
 
     let members = typeDefnNode.Members
+    let hasMembers = List.isNotEmpty members
```

Comment:
I'm not really a fan of this as members can occur twice for certain `TypeDefn`.
[Sample](https://fsprojects.github.io/fantomas-tools/#/oak?data=N4KABGBEDGD2AmBTSAuKAXAngB0WAgmALwA6ATiQHYQTQA2AhgM5PlU0dgC2iXARojJgAHgDoAQsTAAGNtRqJK8ORwDuAS3QALFZwg9%2BgkaIDCUgIyQANOChIAZusqImqKNduR1TAGJN1bvYMdEyINhBeTADK6GSwAK5MsfHYgcGhIAC%2BQA)

Reply:
Are you opposed to the name change here or moving it to the outer scope? Or both 😅?

Reply:
I'm opposed to having the `hasMembers` value on the `genTypeDefn` level (outside the match expression).
That would be confusing as `hasMembers` could be false while the concrete `TypeDefn` still has members. I would move `hasMembers` into the match clauses where you actually use it.
Because of that, there would be some duplication, but at least it would be less ambiguous.